core/src/main/java/com/linecorp/armeria/common/SystemPropertyFlagsProvider.java

```diff
@@ -0,0 +1,419 @@
+        }
+        return CSV_SPLITTER.splitToList(val);
+    }
+
+    @Override
+    public String fileServiceCacheSpec() {
+        return getNormalized("fileServiceCacheSpec");
+    }
+
+    @Override
+    public String dnsCacheSpec() {
+        return getNormalized("dnsCacheSpec");
+    }
+
+    @Override
+    public Predicate<InetAddress> preferredIpV4Addresses() {
+        final String val = getNormalized("preferredIpV4Addresses");
+        if (val == null) {
+            return null;
+        }
+        final List<Predicate<InetAddress>> preferredIpV4Addresses =
+                CSV_SPLITTER.splitToList(val)
+                            .stream()
+                            .map(InetAddressPredicates::ofCidr)
+                            .collect(toImmutableList());
+        if (preferredIpV4Addresses.isEmpty()) {
+            return null;
+        }
+        return preferredIpV4Addresses.stream().reduce(p -> true, Predicate::and);
+    }
+
+    @Override
+    public Boolean useJdkDnsResolver() {
+        return getBoolean("useJdkDnsResolver");
+    }
+
+    @Override
+    public Boolean reportBlockedEventLoop() {
+        return getBoolean("reportBlockedEventLoop");
+    }
+
+    @Override
+    public Boolean validateHeaders() {
+        return getBoolean("validateHeaders");
+    }
+
+    @Override
+    public Boolean tlsAllowUnsafeCiphers() {
+        return getBoolean("tlsAllowUnsafeCiphers");
+    }
+
+    @Override
+    public Set<TransientServiceOption> transientServiceOptions() {
+        final String val = getNormalized("transientServiceOptions");
+        if (val == null) {
+            return null;
+        }
+        return Sets.immutableEnumSet(
+                Streams.stream(CSV_SPLITTER.split(val))
+                       .map(feature -> TransientServiceOption.valueOf(
+                               Ascii.toUpperCase(feature)))
+                       .collect(toImmutableSet()));
+    }
+
+    @Override
+    public Boolean useDefaultSocketOptions() {
+        return getBoolean("useDefaultSocketOptions");
+    }
+
+    @Override
+    public Boolean useLegacyRouteDecoratorOrdering() {
+        return getBoolean("useLegacyRouteDecoratorOrdering");
+    }
+
+    @Override
+    public Boolean allowDoubleDotsInQueryString() {
+        return getBoolean("allowDoubleDotsInQueryString");
+    }
+
+    @Override
+    public Path defaultMultipartUploadsLocation() {
+        return getAndParse("defaultMultipartUploadsLocation", Paths::get);
+    }
+
+    @Nullable
+    private static Long getLong(String name) {
+        return getAndParse(name, Long::parseLong);
+    }
+
+    @Nullable
+    private static Integer getInt(String name) {
+        return getAndParse(name, Integer::parseInt);
+    }
+
+    @Nullable
+    private static Boolean getBoolean(String name) {
+        return getAndParse(name, SystemPropertyFlagsProvider::stickyParseBoolean);
+    }
+
+    private static Boolean stickyParseBoolean(String val) {
```

Comment:
Actually, It was misspelled 😖. It supposed to be `stricklyParseBoolean`. I name it `strickly` because input values have to be either "true" or "false" only otherwise exception is raised, in contrast to normal `Boolean.parseBoolean` that returns false if input isn't "true" ignore case. I copy this behavior from https://github.com/line/armeria/blob/89492ebafc16169af7a9cc9aa07d32b50bf1a673/core/src/main/java/com/linecorp/armeria/common/Flags.java#L1429-L1441.